addon/services/sl-modal.js

```diff
@@ -0,0 +1,96 @@
+import Ember from 'ember';
+
+/**
+ * @module
+ * @augments ember/Service
+ */
+export default Ember.Service.extend({
+
+    /**
+     * Object used to store modal references
+     *
+     * @type {Object}
+     */
+    modals: {},
+
+    /**
+     * Find modal by name
+     *
+     * @function
+     * @param {String} name - The unique name of the component to find
+     * @returns {?ember/Component}
```

Comment:
Believe `get()` returns `undefined` not `null` if provided property doesn't exist.  Please test this and change `@returns` type as appropriate.